specification/loadtestservice/LoadTestService/models.tsp

```diff
@@ -0,0 +1,1049 @@
+
+  @added(APIVersions.v2023_04_01_preview)
+  @doc("If the file is a JSON config file to define the requests for a URL test.")
+  URL_TEST_CONFIG,
+}
+
+@doc("File status.")
+enum FileStatus {
+  @doc("File is not validated.") NOT_VALIDATED,
+  @doc("File is validated.") VALIDATION_SUCCESS,
+  @doc("File validation is failed.") VALIDATION_FAILURE,
+  @doc("File validation is in progress.") VALIDATION_INITIATED,
+  @doc("Validation is not required.") VALIDATION_NOT_REQUIRED,
+}
+
+@doc("Test result based on pass/fail criteria.")
+enum PFTestResult {
+  @doc("Pass/fail criteria has passed.") PASSED,
+  @doc("Pass/fail criteria is not applicable.") NOT_APPLICABLE,
+  @doc("Pass/fail criteria has failed.") FAILED,
+}
+
+@doc("Test run status.")
+enum Status {
+  @doc("Test run request is accepted.") ACCEPTED,
+  @doc("Test run is not yet started.") NOTSTARTED,
+  @doc("Test run is provisioning.") PROVISIONING,
+  @doc("Test run is provisioned.") PROVISIONED,
+  @doc("Test run is getting configured.") CONFIGURING,
+  @doc("Test run configuration is done.") CONFIGURED,
+  @doc("Test run has started executing.") EXECUTING,
+  @doc("Test run execution is completed.") EXECUTED,
+  @doc("Test run is getting deprovisioned.") DEPROVISIONING,
+  @doc("Test run is deprovisioned.") DEPROVISIONED,
+  @doc("Test run is completed.") DONE,
+  @doc("Test run is being cancelled.") CANCELLING,
+  @doc("Test run request is cancelled.") CANCELLED,
+  @doc("Test run request is failed.") FAILED,
+  @doc("Test run JMX file is validated.") VALIDATION_SUCCESS,
+  @doc("Test run JMX file validation is failed.") VALIDATION_FAILURE,
+}
+
+@doc("Aggregation type.")
+enum AggregationType {
+  @doc("Average value.") Average,
+  @doc("Total count.") Count,
+  @doc("Aggregation will be average in this case.") None,
+  @doc("Total sum.") Total,
+  @doc("90th percentile.") Percentile90,
+  @doc("95th percentile.") Percentile95,
+  @doc("99th percentile.") Percentile99,
+}
+
+@doc("Metric unit.")
+enum MetricUnit {
+  @doc("No unit specified.") NotSpecified,
+  @doc("Percentage.") Percent,
+  @doc("Value count.") Count,
+  @doc("Seconds.") Seconds,
+  @doc("Milliseconds") Milliseconds,
+  @doc("Bytes") Bytes,
+  @doc("Bytes per second") BytesPerSecond,
+  @doc("Count per second") CountPerSecond,
+}
+
+@doc("Time Grain")
+enum TimeGrain {
+  @doc("5 seconds, available only if test run duration is less than 10 minutes.")
+  PT5S,
+
+  @doc("10 seconds, available only if test run duration is less than 10 minutes.")
+  PT10S,
+
+  @doc("1 minute")
+  PT1M,
+
+  @doc("5 minutes, available only if test run duration is greater than 1 minute.")
+  PT5M,
+
+  @doc("1 hour, available only if test run duration is greater than 1 minute.")
+  PT1H,
+}
+
+@added(APIVersions.v2023_04_01_preview)
+@doc("Test type")
+enum TestType {
+  @doc("URL Test")
+  URL,
+
+  @doc("JMX Test")
+  JMX,
+}
+
+@doc("Test file info.")
+@resource("files")
+@parentResource(Test)
+model TestFileInfo extends FileInfo {
```

Comment:
I think `is` would be better than `extends` here. You don't really want inheritance. Might also move the fileName into FileIno so it does not have to be declared twice.
```suggestion
model TestFileInfo is FileInfo {
```

Reply:
Thanks for the suggestion, updated the code.